Plugins/NativeEngine/Source/VertexArray.cpp

```diff
@@ -85,7 +95,50 @@ namespace Babylon
 
     void VertexArray::SetVertexBuffers(bgfx::Encoder* encoder, uint32_t startVertex, uint32_t numVertices)
     {
-        for (auto& pair : m_vertexBufferRecords)
+        const bgfx::Caps* caps = bgfx::getCaps();
+
+        // Check if instancing is supported.
+        const bool instancingSupported = 0 != (BGFX_CAPS_INSTANCING & caps->supported);
+        if (!m_vertexBufferInstanceRecords.empty() && instancingSupported)
+        {
+            bgfx::InstanceDataBuffer instanceDataBuffer{};
```

Comment:
I think it would cleaner if this function (that creates a `bgfx::InstanceDataBuffer` from a span of VertexBuffer) is encapsulated in a public static function under VertexBuffer, so that you don't have to expose the bytes with `GetBytes`. It would also make this code a lot easier to read.

Reply:
Done.

Reply:
This isn't exactly what I meant, but we can fix it later.